packages/amplication-git-utils/src/types.ts

```diff
@@ -64,65 +96,13 @@ export interface GitResourceMeta {
   adminUIPath: string;
 }
 
-export interface GitResourceMeta {
-  serverPath: string;
-  adminUIPath: string;
-}
-
 export interface GitClient {
-  createUserRepository(
-    installationId: string,
-    owner: string,
-    name: string,
-    isPublic: boolean
-  ): Promise<RemoteGitRepository>;
-
-  createOrganizationRepository(
-    installationId: string,
-    owner: string,
-    name: string,
-    isPublic: boolean
-  ): Promise<RemoteGitRepository>;
-
-  getOrganizationRepos(
-    installationId: string,
-    limit: number,
-    page: number
-  ): Promise<RemoteGitRepos>;
-
-  isRepoExist(installationId: string, name: string): Promise<boolean>;
-
-  getGitInstallationUrl(workspaceId: string): Promise<string>;
-
-  deleteGitOrganization(installationId: string): Promise<boolean>;
-
-  getGitRemoteOrganization(
-    installationId: string
-  ): Promise<RemoteGitOrganization>;
-
-  getFile(
-    userName: string,
-    repoName: string,
-    path: string,
-    baseBranchName: string,
-    installationId: string
-  ): Promise<GithubFile>;
-
-  createPullRequest(
-    mode: EnumPullRequestMode,
-    userName: string,
-    repoName: string,
-    modules: Required<Changes["files"]>,
-    commitName: string,
-    commitMessage: string,
-    commitDescription: string,
-    installationId: string,
-    head: string
-  ): Promise<string>;
-
-  getRepository(
-    installationId: string,
-    owner: string,
-    repo: string
-  ): Promise<RemoteGitRepository>;
+  getRepository(options: Options): Promise<RemoteGitRepository>;
+  getRepositories(options: Options): Promise<RemoteGitRepos>;
+  createRepository(options: Options): Promise<RemoteGitRepository>;
+  deleteGitOrganization(options: Options): Promise<boolean>;
+  getGitRemoteOrganization(options: Options): Promise<RemoteGitOrganization>;
+  getFile(options: Options): Promise<GithubFile>;
+  createPullRequest(options: Options): Promise<string>;
+  getGitInstallationUrl(options: Options): Promise<string>;
```

Comment:
Each function should use its own parameters, it will be very hard to use the lib if I need to provide all parameters for every type of operation, without knowing which parameter is needed to which operation

Reply:
I agree, but there are differences between the providers and if one parameter is mandatory in provider X and not in providers Y and Z, we will still have to set some parameters as optional although it is mandatory in one provider, and this is also confusing.